dts: vendor/arduino: fix QSPI partitioning

pillo79 opened the pull request:
The Wi-Fi firmware image in the QSPI flash devices has always been located at 512kB from the end of the 16MB Flash device. This offset is 0xf80000, not 0xe80000 as it was mistakenly set.

Also, add the definition for the Master Boot Record partition at the beginning of the flash. This is used by existing Arduino 'Storage' libraries to store partition table information.

soburi reviewed (commented):
> The Wi-Fi firmware image in the QSPI flash devices has always been located at 512kB from the end of the 16MB Flash device. This offset is 0xf80000, not 0xe80000 as it was mistakenly set.

> Also, add the definition for the Master Boot Record partition at the beginning of the flash. This is used by existing Arduino 'Storage' libraries to store partition table information.

Are these issues independent of each other?
If so, at least separating the commit is better.

pillo79 committed:
dts: vendor/arduino: add missing 'mbr' partition

Add the definition for the Master Boot Record partition at the
beginning of the flash. This is used by existing Arduino 'Storage'
libraries to store partition table information.

Signed-off-by: Luca Burelli <l.burelli@arduino.cc>

pillo79 committed:
dts: vendor/arduino: fix firmware partition location

The Wi-Fi firmware image in the QSPI flash has always been located in
the last 512kB of the 16MB Flash device. This offset is 0xf80000, not
0xe80000 as it was mistakenly set. Fix.

Signed-off-by: Luca Burelli <l.burelli@arduino.cc>
